go/src/koding/kites/kloud/provider/builder.go

```diff
@@ -311,13 +311,33 @@ func (bs *BaseStack) applyAsync(ctx context.Context, req *kloud.ApplyRequest) er
 		return err
 	}
 
+	// At this point resources have been built successfully on the terraform side.
+	// If updating stack or provisioning klient on remote hosts fails,
+	// we need to cleanup the stack; if debug: true was set in the stack template
+	// for any of the instance resources, the stack won't be destroyed
+	// so it's possible to troubleshoot it post-mortem.
+	if !bs.Debug {
+		defer func() {
+			if err != nil {
+				_, e := tfKite.Destroy(&tf.TerraformRequest{
```

Comment:
isnt this problematic if the admin is in the progress of incremential updates? how can we decide on behalf of user to destroy the stack template? even terraform does not automatically destroy the created resources and warns the user to manually edit created resources, i see many problems with this approach and advise to revert it. Is there anything that i am missing?